sockeye/beam_search.py

```diff
@@ -102,7 +104,6 @@ def decode_step(self,
         outputs, new_states = [], []
         for model, model_states in zip(self._models, states):
             logits, model_states, _ = model.decode_step(step_input, model_states, vocab_slice_ids)
-            logits = logits.astype('float32', copy=False)
```

Comment:
Actually, looking at the TODO above, the cast is not something that's needed anymore since I've made changes to the rest of beam search so that the dtype matches the model's dtype. I'll remove the TODO regarding that.